lib/_emerge/depgraph.py

```diff
@@ -5169,6 +5170,11 @@ def _resolve(self, myfavorites):
                     )
                     raise
 
+        try:
+            del myroot, pkgsettings, pprovideddict, virtuals
+        except NameError:
+            pass
+
```

Comment:
Maybe do this in the scope of the loop, so that we don't have to ignore the NameError.

Reply:
I was going to, but there's a couple of `break`s in there. Someone else suggested adding `del` for safety, but they hadn't considered what would happen if it didn't enter the loop, so then I added the `try`.